### PR TITLE
Add compatibility with Haste 5

### PR DESCRIPTION
**Pull request by qzminski**

### CODE REVIEW

File: classes/DcaWizardHelper.php [hunk 1]
```diff
@@ -38,55 +40,37 @@ public function handleAjaxActions($strAction, \DataContainer $dc)
             if ('File' === $GLOBALS['TL_DCA'][$dc->table]['config']['dataContainer']) {
                 // The field does not exist
                 if (!array_key_exists($strField, $GLOBALS['TL_CONFIG'])) {
-                    \System::log(
-                        'Field "' . $strField . '" does not exist in the global configuration',
-                        __METHOD__,
-                        TL_ERROR
-                    );
-
                     header('HTTP/1.1 400 Bad Request');
```
Comment (Toflar) [inline classes/DcaWizardHelper.php:43]: Should probably be a response exception?

File: classes/DcaWizardHelper.php [hunk 2]
```diff
@@ -38,55 +40,37 @@ public function handleAjaxActions($strAction, \DataContainer $dc)
...
+            } elseif (Database::getInstance()->tableExists($dc->table)) {
                 // The field does not exist
                 if (!isset($GLOBALS['TL_DCA'][$dc->table]['fields'][$strField])) {
-                    \System::log(
-                        'Field "' . $strField . '" does not exist in table "' . $dc->table . '"',
-                        __METHOD__,
-                        TL_ERROR
-                    );
-
                     header('HTTP/1.1 400 Bad Request');
```
Comment (Toflar) [inline classes/DcaWizardHelper.php:50]: Should probably be a response exception?

File: classes/DcaWizardHelper.php [hunk 3]
```diff
@@ -38,55 +40,37 @@ public function handleAjaxActions($strAction, \DataContainer $dc)
...
 
                 // The record does not exist
                 if (!$objRow->numRows) {
-                    \System::log(
-                        'A record with the ID "' . $intId . '" does not exist in table "' . $dc->table . '"',
-                        __METHOD__,
-                        TL_ERROR
-                    );
-
                     header('HTTP/1.1 400 Bad Request');
```
Comment (Toflar) [inline classes/DcaWizardHelper.php:61]: Should probably be a response exception?

File: classes/DcaWizardHelper.php [hunk 4]
```diff
@@ -101,95 +85,14 @@ public function handleAjaxActions($strAction, \DataContainer $dc)
...
             }
 
-            /** @var \Widget $objWidget */
+            /** @var Widget $objWidget */
             $objWidget = new $strClass(
                 $strClass::getAttributesFromDca($arrData, $strFieldName, null, $strField, $dc->table, $dc)
             );
 
-            header('Content-Type: text/html; charset=' . $GLOBALS['TL_CONFIG']['characterSet']);
+            header('Content-Type: text/html; charset=' . \Contao\Config::get('characterSet'));
```
Comment (Toflar) [inline classes/DcaWizardHelper.php:93]: Should probably be a response exception?

**Review by Toflar — CHANGES_REQUESTED** [review]

**Comment by Toflar** [comment]
Also there is a conflict to resolve :)

**Comment by qzminski** [comment]
Bugfix or minor version?

**Review by Toflar — APPROVED** [review]
Minor :)

File: widgets/DcaWizard.php [hunk 5]
```diff
@@ -302,9 +302,8 @@ public function getDcaWizardOptions()
      */
     public function getButtonHref()
     {
-        return ampersand(\Environment::get('base')
-            . TL_SCRIPT
-            . '?'
+        return StringUtil::ampersand(Environment::get('base')
```
Comment (aschempp) [inline widgets/DcaWizard.php:305]: can we generate a real route here? Because the `/contao` prefix can be changed in the configuration

Comment (qzminski) [reply widgets/DcaWizard.php:305]: Changed in 42b782f.

**Pull request merged**